scripts/zones/Norg/npcs/Stray_Cloud.lua

```diff
@@ -1,18 +1,58 @@
 -----------------------------------
 -- Area: Norg
 --  NPC: Stray Cloud
--- Standard Info NPC
+-- Starts and Ends Quest: An Undying Pledge
+-- !pos-20.617, 1.097, -29.165, 133
 -----------------------------------
+require("scripts/globals/titles")
+require("scripts/globals/settings")
+require("scripts/globals/keyitems")
+require("scripts/globals/quests")
+local ID = require("scripts/zones/Norg/IDs")
+-----------------------------------
+
 
 function onTrade(player, npc, trade)
 end
 
 function onTrigger(player, npc)
-    player:startEvent(231)
-end
+    Pledge = player:getQuestStatus(OUTLANDS, tpz.quest.id.outlands.AN_UNDYING_PLEDGE) 	
+
+    if (Pledge == QUEST_AVAILABLE and player:getFameLevel(NORG) >= 4) then
```

Comment:
project standard is to not include parens unless needed, so
`if Pledge == QUEST_AVAILABLE and player:getFameLevel(NORG) >= 4 then`